### DISCUSSION TITLE
[Usage question] - What makes a mock session being uploaded to FI

### DISCUSSION THREAD

**Discussion by GaelleChapuis** [int-brain-lab/iblenv#133] [Q&A]
**Describe the question you have**
Current instructions for running a mock session (on Ephys rig) are:

- Use the subject name “iblrig_test_mouse”, and follow the steps:
- Put in a probe
- Prepare an ephys and video session for that subject name
- Start spikeGLX
- Run ephysChoiceWorld and select mock
- Do a couple trials with your fingers on the wheel ; Minimum 42 trials otherwise the transfer_*_session script would not run
- Stop the recording as you would normally for a real recording session

In other words, if a mock session has >42 trials, it will be uploaded to FI - correct ?

Example of recent mock session uploaded to FI: 
```
eid  = 13bd02ba-d8ba-4da0-97ec-fba997ddc9ab
sess_det = one.get_details(eid='13bd02ba-d8ba-4da0-97ec-fba997ddc9ab', full=True)
```
`sess_det['json']['IS_MOCK']` is `True`
`sess-det['n_trials]` is `574` (@alejandropan did you do so many trials with your fingers?!)

**Comment by alejandropan** [comment]
@GaelleChapuis I think I have a misunderstanding about mock recordings. The way I have been doing mock recordings is by running a mouse the day before the actual recordings with electrodes outside the brain but still acquiring (hence the high number of trials).  Is this not what we are supposed to do?

**Comment by GaelleChapuis** [comment]
There are two cases, I will clarify here:

1) there is a "mock Ephys" session to be done one day before starting recording with the mouse in the Ephys rig, mimicking all the steps of a real recording. This data should be uploaded and registered.

2) here I was discussing a different kind of mock session: one that a user would do with its own fingers to quickly check the rig is working. This data should not be uploaded.

At present, I am not sure how to distinguish those on the database - probably by the nickname of the subject? If it is not `test_mouse`, it would be a "mock but real" session as per pt 1)?

**Comment by nbonacchi** [comment]
The second case should not be on Alyx at all, in fact if the subject is _iblrig_test_mouse it will not register. If you find such sessions maybe they are very old? How many are there? I recall that at some point we allowed _iblrig_test_mouse to register. Can you check?
